### PR TITLE
Remove CPhase + Tidy up src

### PR DESCRIPTION
**Pull request by maliasadi**
- Remove `CPhase` in favour of `CPhaseShift` in Lightning devices. [qml.CPhase](https://docs.pennylane.ai/en/stable/code/api/pennylane.CPhase.html) is an alias for `qml.ControlledPhaseShift` and can be removed from the the list of supported ops and toml files.
- Update the name of CI C++ and Python tests to include the matrix strategy info in the list of required actions.
- Tidy up a few files and scripts.

[sc-62667]
[sc-58634]

### CODE REVIEW

**Review by dime10 — APPROVED** [review]
Thanks Ali!

File: pennylane_lightning/core/src/utils/NDPermuter.hpp [hunk 1]
```diff
@@ -242,7 +244,7 @@ template <size_t BLOCKSIZE = 1024> class DefaultPermuter {
 
                 bool complete{true};
                 // NOLINTBEGIN
-                for (size_t j = num_indices; j--;) {
+                for (std::size_t j = num_indices; j--;) {
```
Comment (multiphaseCFD) [inline pennylane_lightning/core/src/utils/NDPermuter.hpp:247]: Is `j==0` the stop condition here?

File: pennylane_lightning/core/src/utils/NDPermuter.hpp [hunk 2]
```diff
@@ -189,11 +191,11 @@ template <size_t BLOCKSIZE = 1024> class DefaultPermuter {
...
         }
 
-        std::vector<size_t> old_super_dimensions(num_indices, 1);
-        std::vector<size_t> new_super_dimensions(num_indices, 1);
+        std::vector<std::size_t> old_super_dimensions(num_indices, 1);
+        std::vector<std::size_t> new_super_dimensions(num_indices, 1);
 
         const std::size_t old_dimensions_size = shape.size();
-        for (size_t i = old_dimensions_size; --i;) {
+        for (std::size_t i = old_dimensions_size; --i;) {
```
Comment (multiphaseCFD) [inline pennylane_lightning/core/src/utils/NDPermuter.hpp:198]: Do we need `;` here after `--i`?

File: pennylane_lightning/core/src/simulators/lightning_gpu/measurements/MeasurementsGPU.hpp [hunk 3]
```diff
@@ -15,7 +15,7 @@
 /**
  * @file
  * Defines a class for the measurement of observables in quantum states
- * represented by a Lightning Qubit StateVector class.
+ * represented by a Lightning GPU StateVector class.
```
Comment (multiphaseCFD) [inline pennylane_lightning/core/src/simulators/lightning_gpu/measurements/MeasurementsGPU.hpp:18]: Nice catch!

**Review by multiphaseCFD — APPROVED** [review]
Thanks @maliasadi ! LGTM!

File: Makefile [hunk 4]
```diff
@@ -89,8 +89,8 @@ test-builtin:
 	PL_DEVICE=$(if $(device:-=),$(device),lightning.qubit) $(PYTHON) -I $(TESTRUNNER)
 
 test-suite:
-	pl-device-test --device $(if $(device:-=),$(device),lightning.qubit) --skip-ops --shots=20000
-	pl-device-test --device $(if $(device:-=),$(device),lightning.qubit) --shots=None --skip-ops
+	pl-device-test --device $(if $(device:-=),$(device),lightning.qubit) --skip-ops --shots=20000 -n auto
+	pl-device-test --device $(if $(device:-=),$(device),lightning.qubit) --shots=None --skip-ops -n auto
```
Comment (vincentmr) [inline Makefile:93]: `test-suite` is quite fast, so I don't know whether it is worth forcing a dependency on `pytest-xdist`, but I don't mind either way.

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -6,10 +6,16 @@
 
 ### Improvements
 
+* Update C++ and Python GitHub actions names to include the matrix info.
+  [(#717)](https://github.com/PennyLaneAI/pennylane-lightning/pull/717)
+
```
Comment (vincentmr) [inline .github/CHANGELOG.md:11]: I like this improvement, we should have more systematic naming in the CI. But why was this necessary in this PR?

File: pennylane_lightning/core/src/utils/NDPermuter.hpp [hunk 6]
```diff
@@ -68,7 +68,8 @@ template <class PermuterBackend> class Permuter {
      */
     template <class T>
     void Transpose(const std::vector<T> &data_in,
-                   const std::vector<size_t> &shape, std::vector<T> &data_out,
```
Comment (vincentmr) [inline pennylane_lightning/core/src/utils/NDPermuter.hpp:71]: Like that too. I think we should even make a `size_t` => `std::size_t` PR and be rid of it.

**Review by vincentmr — APPROVED** [review]
A couple questions, but ready to approve ... so I'll approve. Thanks @maliasadi , nice one.

File: pennylane_lightning/core/src/utils/NDPermuter.hpp (hunk 1 above)
Comment (maliasadi) [reply pennylane_lightning/core/src/utils/NDPermuter.hpp:247]: The stop condition is `++old_counter[j] < shape[j]` or `j==0` whichever satisfies sooner.

File: .github/CHANGELOG.md (hunk 5 above)
Comment (maliasadi) [reply .github/CHANGELOG.md:11]: This is required to define proper CI requirements for PRs.

File: pennylane_lightning/core/src/utils/NDPermuter.hpp (hunk 2 above)
Comment (maliasadi) [reply pennylane_lightning/core/src/utils/NDPermuter.hpp:198]: Yes, we need this as we don't have any explicit stopping condition here.

**Pull request merged**